navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/LocationUpdatedThread.java

```diff
@@ -1,349 +0,0 @@
-                             final Location location) {
-    if (location == null || directionsRoute == null) {
-      return;
-    }
-
-    Timber.d("Previous route alert level: %d", previousRouteProgress.getAlertUserLevel());
-
-    // With a new location update, we create a new RouteProgress object.
-    final RouteProgress routeProgress = monitorStepProgress(previousRouteProgress, location);
-
-    userOffRoute = userIsOnRoute(location, routeProgress.getCurrentLeg());
-
-    if (snapToRoute && !userOffRoute) {
-      // Pass in the snapped location with all the other location data remaining intact for their use.
-      location.setLatitude(routeProgress.usersCurrentSnappedPosition().getLatitude());
-      location.setLongitude(routeProgress.usersCurrentSnappedPosition().getLongitude());
-
-      location.setBearing(snapUserBearing(routeProgress));
-    }
-
-    this.location = location;
-
-    // Post back to the UI Thread.
-    responseHandler.post(new Runnable() {
-      public void run() {
-
-        // Only report user off route once.
-        if (userOffRoute && (userOffRoute != previousUserOffRoute)) {
-          for (OffRouteListener offRouteListener : offRouteListeners) {
-            offRouteListener.userOffRoute(location);
-          }
-          previousUserOffRoute = userOffRoute;
-        }
-
-        if (previousRouteProgress.getAlertUserLevel() != routeProgress.getAlertUserLevel()) {
-
-          for (AlertLevelChangeListener alertLevelChangeListener : alertLevelChangeListeners) {
-            alertLevelChangeListener.onAlertLevelChange(routeProgress.getAlertUserLevel(), routeProgress);
-          }
-        }
-
-        for (ProgressChangeListener progressChangeListener : progressChangeListeners) {
-          progressChangeListener.onProgressChange(LocationUpdatedThread.this.location, routeProgress);
-        }
-      }
-    });
-  }
-
-  private RouteProgress monitorStepProgress(@NonNull RouteProgress routeProgress, Location location) {
-    int currentStepIndex = routeProgress.getCurrentLegProgress().getStepIndex();
-    int currentLegIndex = routeProgress.getLegIndex();
-
-    // Force an announcement when the user begins a route
-    int alertLevel = routeProgress.getAlertUserLevel() == NavigationConstants.NONE_ALERT_LEVEL
-      ? NavigationConstants.DEPART_ALERT_LEVEL : routeProgress.getAlertUserLevel();
-
-    Position truePosition = Position.fromCoordinates(location.getLongitude(), location.getLatitude());
-    double userSnapToStepDistanceFromManeuver = RouteUtils.getDistanceToNextStep(
-      truePosition,
-      routeProgress.getCurrentLeg(),
-      routeProgress.getCurrentLegProgress().getStepIndex()
-    );
-
-    double secondsToEndOfStep = userSnapToStepDistanceFromManeuver / location.getSpeed();
-    boolean courseMatchesManeuverFinalHeading = false;
-
-    // TODO set to eventually adjust for different direction profiles.
```

Comment:
support for different direction profiles is already ticked in https://github.com/mapbox/mapbox-navigation-android/issues/24